gitlab.go

```diff
@@ -497,6 +497,34 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 	return response, err
 }
 
+// Lint validates .gitlab-ci content
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/lint.html
+func (c *Client) Lint(content string, options ...OptionFunc) (valid bool, err error) {
+	if content == "" {
+		return false, errors.New("content is missing")
+	}
+	opts := struct {
+		Content string `json:"content"`
+	}{
+		Content: content,
+	}
+	req, err := c.NewRequest("POST", "ci/lint", opts, options)
+	if err != nil {
+		return false, err
+	}
+
+	lint := struct {
+		Status string   `json:"status"`
+		Errors []string `json:"errors"`
+	}{}
+	_, err = c.Do(req, &lint)
+	if err != nil {
+		return false, err
+	}
+	return lint.Status == "valid", nil
```

Comment:
I think we should just return the whole `lint` struct here, so people have access to both the status and any possible errors.

Reply:
It looks like `client.Do` takes care of marshaling `error []string`  into an error type and would get returned above. I'll change it, just trying to check my understanding of the package.

Reply:
IIRC, I was actually able to observe this in testing. It seems like the anonymous struct above really only needs the status and the caller only needs to know if its valid or not since if their are errors `client.Do` will return them.

Reply:
No, I don't think so... The errors in this struct are errors in your `ci` file. The errors returned by `client.Do` are errors caused by functions that failed to execute as expected.

Reply:
@svanharmelen Perhaps I'm reading this wrong, it looks like https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L529 is doing just that?

Reply:
Yes, I think your reading it wrong :wink: The errors in the `Lint` response struct are syntax errors in your CI file. The errors the `client` processes are HTTP or server errors (https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L530-L533).

When your CI file contains errors, I expect that the HTTP call will still return a 200 as the HTTP call was successful. Didn't test that last part, but that should be easy to so with a curl...